include/libdnf/conf/config.hpp

```diff
@@ -20,15 +20,49 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #ifndef LIBDNF_CONF_CONFIG_HPP
 #define LIBDNF_CONF_CONFIG_HPP
 
+#include "config_parser.hpp"
 #include "option_binds.hpp"
+#include "option.hpp"
+#include "vars.hpp"
+#include "../logger/logger.hpp"
+
+#include <fmt/format.h>
 
 namespace libdnf {
 
 /// Base class for configurations objects
+template <Option::Priority default_priority>
```

Comment:
Does it have to be a template? What about an internal variable, adjustable e.g. in the constructor. Yes, each instance will be larger by this variable. But it saves a piece of code. And again there will be one class `Config` from which the others will inherit. Instead of multiple parent classes generated from a template.